modules/Bio/EnsEMBL/IO/Translator/StructuralVariationFeature.pm

```diff
@@ -0,0 +1,339 @@
+
+=cut
+
+package Bio::EnsEMBL::IO::Translator::StructuralVariationFeature;
+
+use strict;
+use warnings;
+
+use Carp;
+
+use base qw/Bio::EnsEMBL::IO::Translator::Feature/;
+
+my %field_callbacks = ('chrom'  => 'chrom',
+                       'pos'    => 'pos',
+                       'id'     => 'id',
+                       'ref'    => 'ref',
+                       'alt'    => 'alt',
+                       'qual'   => 'qual',
+                       'filter' => 'filter',
+                       'info'   => 'info'
+                      );
+
+my %info_methods = ('SVTYPE'    => 'sv_type',
+                    'END'       => 'sv_end',
+                    'Parent'    => 'sv_parent',
+                    'SVLEN'     => 'sv_length',
+                    'CIPOS'     => 'sv_cipos',
+                    'CIEND'     => 'sv_ciend',
+                    'IMPRECISE' => 'sv_imprecise'
+                   );
+
+my %svtypes = ('copy_number_variation' => 'CNV',
+               'copy_number_loss' => 'CNV:LOSS',
+               'copy_number_gain' => 'CNV:GAIN',
+               'deletion' => 'DEL',
+               'inversion' => 'INV',
+               'insertion' => 'INS',
+               'novel_sequence_insertion' => 'INS',
+               'mobile_element_insertion' => 'INS:ME',
+               'Alu_insertion' => 'INS:ME:ALU',
+               'duplication' => 'DUP',
+               'tandem_duplication' => 'DUP:Tandem',
+        
+               'sequence_alteration' => 'SA',
+               'complex_structural_alteration' => 'CSA',
+               'complex_substitution' => 'CS',
+               'indel' => 'INDEL',
+               'loss_of_heterozygosity' => 'LOH',
+               'translocation' => 'TL',
+               'interchromosomal_translocation' => 'TL:Inter',
+               'intrachromosomal_translocation' => 'TL:Intra',
+               'interchromosomal_breakpoint' => 'BP:InterC',
+               'intrachromosomal_breakpoint' => 'BP:IntraC',
+              );                            
+#my %info_methods = ('AA' => 'ancestral_allele',
+#                    #'AC'  => 'allele_count',
+#                    #'AF'  => 'allele_frequency',
+#                    #'AN'  => 'total_allele_number',
+#                    'MA'  => 'minor_allele',
+#                    'MAF' => 'minor_allele_frequency',
+#                    'MAC' => 'minor_allele_count',
+#                    'NS'  => 'samples_with_data'
+#                   );
+
+=head2 new
+
+    Returntype   : Bio::EnsEMBL::IO::Translator::SampleGenotypeFeature
+
+=cut
+
+sub new {
+    my ($class, $info_list, $formats_list, $samples_list) = @_;
+  
+    my $self = $class->SUPER::new();
+
+    $samples_list = [] if (!$samples_list);
+    $self->{'samples_list'} = $samples_list;
+
+    if (scalar(@$samples_list) > 0) {
+      $field_callbacks{'format'} = 'format';
+    }
+    else {
+      warn "There is no individuals/samples list provided!\n";
```

Comment:
Reword here too. "No individuals/samples list provided", or "There is no list of samples provided". Maybe change to a croak while you're at it.